### ISSUE TITLE
the result of validation is different from result on the end of training

### ISSUE THREAD

**Issue by fppccc** [ultralytics/ultralytics#1198]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello,I trained YOLOv8  on my datasets,and used code as below: 

'''------------'''
from ultralytics import YOLO
model = YOLO("yolov8m.yaml")  # build a new model from scratch
results = model.train(data="coco_mydata.yaml", epochs=200)  # train the model
results = model.val()  # evaluate model performance on the validation set
'''------------'''

But the result of validation（ model.val() ） is different from the result on the end of training.It's confusing.Because they are both calculated on the validation datasets with best.pt,why are they different?
**Which one should I choose to write my paper?**
Thank you very much!

**Comment by glenn-jocher** [comment]
@fppccc results seem almost identical in my test in Colab:
<img width="1213" alt="Screenshot 2023-03-04 at 12 51 40" src="https://user-images.githubusercontent.com/26833433/222928426-778384f3-cdaa-4550-9f96-7c8b6ed23636.png">
